Frameworks/AddressBook/ABRecord.mm

```diff
@@ -187,26 +184,188 @@ CFTypeRef ABRecordCopyValue(ABRecordRef record, ABPropertyID contactProperty) {
         // CFDateRef
     } else if (contactProperty == kABPersonModificationDateProperty) {
         // CFDateRef
-    }
-
-    // TODO #678
-    // Cases for various multi-value properties.
-    if (contactProperty == kABPersonEmailProperty) {
-        // ABMultiValueRef of CFStringRef
-    } else if (contactProperty == kABPersonAddressProperty) {
-        // ABMultiValueRef of CFDictionaryRef
-    } else if (contactProperty == kABPersonDateProperty) {
-        // ABMultiValueRef of CFDateRef
-    } else if (contactProperty == kABPersonPhoneProperty) {
-        // ABMultiValueRef of CFStringRef
     } else if (contactProperty == kABPersonInstantMessageProperty) {
         // ABMultiValueRef of CFDictionaryRef
     } else if (contactProperty == kABPersonSocialProfileProperty) {
         // ABMultiValueRef of CFDictionaryRef
-    } else if (contactProperty == kABPersonURLProperty) {
-        // ABMultiValueRef of CFStringRef
-    } else if (contactProperty == kABPersonRelatedNamesProperty) {
-        // ABMultiValueRef of CFStringRef
+    }
+
+    // Cases for various multi-value properties.
+    if (contactProperty == kABPersonEmailProperty) {
+        _ABMultiValue* emailMultiValue = [[_ABMultiValue alloc] initWithPropertyType:kABStringPropertyType];
+        NSArray* emails = person.contact.emails;
+        [emails enumerateObjectsUsingBlock:^void(id obj, NSUInteger idx, BOOL* stop) {
+            WACContactEmail* email = (WACContactEmail*)obj;
+            CFStringRef label;
+            switch (email.kind) {
+                case WACContactEmailKindPersonal:
+                    label = kABHomeLabel;
+                    break;
+                case WACContactEmailKindWork:
+                    label = kABWorkLabel;
+                    break;
+                case WACContactEmailKindOther:
+                    label = kABOtherLabel;
+                    break;
+            }
+
+            [emailMultiValue appendPairWithLabel:(__bridge NSString*)label andValue:email.address];
+        }];
+
+        [emailMultiValue makeImmutable];
+        return (__bridge_retained CFTypeRef)emailMultiValue;
```

Comment:
I figured it would be easier and be less code to just have the flag for being mutable or not

---
In reply to: [75182693](https://github.com/Microsoft/WinObjC/pull/884#discussion_r75182693) [](ancestors = 75182693)